src/agent0/chainsync/db/hyperdrive/convert_data.py

```diff
@@ -6,107 +6,392 @@
+        as_shares_idx = ~events_df["asBase"] & events_idx
+        events_df.loc[as_base_idx, "base_delta"] = events_df.loc[as_base_idx, "amount"].apply(
+            lambda x: Decimal(x) / Decimal(1e18)  # type: ignore
+        )
+        events_df.loc[as_shares_idx, "vault_share_delta"] = events_df.loc[as_shares_idx, "amount"].apply(
+            lambda x: Decimal(x) / Decimal(1e18)  # type: ignore
+        )
+
+    # Shorts
+    events_idx = events_df["event"].isin(["OpenShort", "CloseShort"])
+    if events_idx.any():
+        events_df.loc[events_idx, "token_type"] = "SHORT"
+        events_df.loc[events_idx, "token_id"] = "SHORT-" + events_df.loc[events_idx, "maturityTime"].astype(int).astype(
+            str
+        )
+
+    events_idx = events_df["event"] == "OpenShort"
+    if events_idx.any():
+        # Pandas apply doesn't play nice with types
+        events_df.loc[events_idx, "token_delta"] = events_df.loc[events_idx, "bondAmount"].apply(
+            lambda x: Decimal(x) / Decimal(1e18)  # type: ignore
+        )
+        as_base_idx = events_df["asBase"] & events_idx
+        as_shares_idx = ~events_df["asBase"] & events_idx
+        events_df.loc[as_base_idx, "base_delta"] = -events_df.loc[as_base_idx, "amount"].apply(
+            lambda x: Decimal(x) / Decimal(1e18)  # type: ignore
+        )
+        events_df.loc[as_shares_idx, "vault_share_delta"] = -events_df.loc[as_shares_idx, "amount"].apply(
+            lambda x: Decimal(x) / Decimal(1e18)  # type: ignore
+        )
+
+    events_idx = events_df["event"] == "CloseShort"
+    if events_idx.any():
+        # Pandas apply doesn't play nice with types
+        events_df.loc[events_idx, "token_delta"] = -events_df.loc[events_idx, "bondAmount"].apply(
+            lambda x: Decimal(x) / Decimal(1e18)  # type: ignore
+        )
+        as_base_idx = events_df["asBase"] & events_idx
+        as_shares_idx = ~events_df["asBase"] & events_idx
+        events_df.loc[as_base_idx, "base_delta"] = events_df.loc[as_base_idx, "amount"].apply(
+            lambda x: Decimal(x) / Decimal(1e18)  # type: ignore
+        )
+        events_df.loc[as_shares_idx, "vault_share_delta"] = events_df.loc[as_shares_idx, "amount"].apply(
+            lambda x: Decimal(x) / Decimal(1e18)  # type: ignore
+        )
+
+    # Add solo transfer events to events_df
+    events_df = pd.concat([events_df, transfer_events_df], axis=0)
+
+    # We select the subset of columns we need and rename to match db schema
+    rename_dict = {
+        "address": "hyperdrive_address",
+        "transactionHash": "transaction_hash",
+        "blockNumber": "block_number",
+        "trader": "wallet_address",
+        "event": "event_type",
+        "token_type": "token_type",
+        "maturityTime": "maturity_time",
+        "token_id": "token_id",
+        "token_delta": "token_delta",
+        "base_delta": "base_delta",
+        "vault_share_delta": "vault_share_delta",
+        "asBase": "as_base",
```

Comment:
Like before, I think `base_delta` is not always in units of `base`, but rather is base or shares depending on `as_base`.